src/latexstyleparser.cpp

```diff
@@ -736,28 +734,18 @@ QStringList LatexStyleParser::readPackageTexDef(QString fn) const
 		return QStringList();
 
 	QString fname = fn.left(fn.length() - 4);
-    QProcess myProc(nullptr);
-
-	//add exec search path
+	ExecProgram execProgram;
+	execProgram.program = texdefDir + "texdef";
+	execProgram.arguments << "-t" << "latex" << "-l" << "-p" << fname;
 	if (!texdefDir.isEmpty()) {
-		QProcessEnvironment env = QProcessEnvironment::systemEnvironment();
-		if (env.value("SHELL") == "/bin/bash") {
-			env.insert("PATH", env.value("PATH") + ":" + texdefDir);
-			myProc.setProcessEnvironment(env);
-		}
+		execProgram.additionalSearchPaths = texdefDir;
 	}
-	QStringList args;
-	QString result;
-	args << "-t" << "latex" << "-l" << "-p" << fname;
-	myProc.start(texdefDir + "texdef", args);
-	args.clear();
-	myProc.waitForFinished();
-	if (myProc.exitCode() != 0)
+	execProgram.execAndWait ();
+	if (!execProgram.normalRun || (execProgram.exitCode != 0)) {
```

Comment:
to my taste, a method like runFinishedWithoutError could simply things here (and later)

Reply:
OK, changed it so that `execAndWait` effectively returns 
`normalRun && (execProgram.exitCode == 0)`
So now we are checking the result when calling `execProgram.execAndWait`